### PR TITLE
Fixed: `highlights_for` would fail for books with large numbers of highlights

### PR DESCRIPTION
**Pull request by phillco**
This endpoint likes to throw 500 exceptions for books with large numbers of highlights (~700 in my experience). Some testing reveals batches over 250 or so to be unreliable, so I chose 200 as the default batch size.

Also theoretically support > 1000 highlights per book.

### CODE REVIEW

File: lib/kindle_highlights/client.rb [hunk 1]
```diff
@@ -12,9 +15,20 @@ def initialize(email_address, password)
     end
   
     def highlights_for(asin)
-      highlights = @mechanize_agent.get("https://kindle.amazon.com/kcw/highlights?asin=#{asin}&cursor=0&count=1000")
-      json       = JSON.parse(highlights.body)
-      json["items"]
+      cursor = 0
+      highlights = []
+
+      while true
```
Comment (speric) [inline lib/kindle_highlights/client.rb]: Do you think it'd be more idiomatic to do:

``` ruby
loop do
  # get items
  break unless json['items'] and json['items'].length > 0
  # move cursor, etc.
end
```

**Comment by speric** [comment]
Thanks for opening this PR! I left a small bit of feedback.

File: lib/kindle_highlights/client.rb (hunk 1 above)
Comment (phillco) [inline lib/kindle_highlights/client.rb]: Nice! I don't actually write Ruby all that often :D. I'll make the change.

**Comment by speric** [comment]
@phillco would you mind squashing these commits down? Thanks!

**Comment by phillco** [comment]
@speric done, sorry for the delay!

**Pull request merged**